### PR TITLE
refactor: Use experimental threads, flow and models mode by default

### PR DESCRIPTION
**Pull request by chamini2**
## TODO
- [x] fix tests to expect flow mode
- [x] fix tests to expect models mode

<details>
closes fea-261
closes fea-262
</details>

### CODE REVIEW

**Review by isidentical — APPROVED** [review]

**Review by turbo1912 — APPROVED** [review]

File: integration_tests/features/environment.py [hunk 1]
```diff
@@ -21,3 +24,9 @@ def after_scenario(context, scenario):
         print_exception(*context.exc)
         _etype, exception, _tb = context.exc
         raise exception
+
+
+def before_tag(context, tag):
```
Comment (isidentical) [inline integration_tests/features/environment.py:29]: ATL: Where do we use this function?

File: integration_tests/features/steps/fal_steps.py [hunk 2]
```diff
@@ -108,11 +108,23 @@ def invoke_command(context):
...
+    # from io import StringIO
+    # log_cap: LoggingCapture = context.log_capture
+    # out_cap: StringIO = context.stdout_capture
+    # err_cap: StringIO = context.stderr_capture
+
+    if context.exc:
+        _etype, exception, _tb = context.exc
+        assert isinstance(
+            exception, eval(etype)
+        ), f"Invalid exception - expected {type}, got {type(exception)}"
```
Comment (isidentical) [inline integration_tests/features/steps/fal_steps.py:123]: Isn't this `etype`?
```suggestion
        ), f"Invalid exception - expected {etype}, got {type(exception)}"
```

**Review by isidentical — APPROVED** [review]

File: integration_tests/features/environment.py (hunk 1 above)
Comment (chamini2) [reply integration_tests/features/environment.py:29]: It's used by behave: https://behave.readthedocs.io/en/stable/api.html?highlight=before_tag#environment-file-functions

Comment (isidentical) [reply integration_tests/features/environment.py:29]: thanks!

**Pull request merged**